src/main/java/org/springframework/data/redis/connection/jedis/JedisConnectionFactory.java

```diff
@@ -963,76 +993,136 @@ private Jedis getActiveSentinel() {
 		throw new InvalidDataAccessResourceUsageException("No Sentinel found");
 	}
 
+	private boolean isCreatedOrStopped(@Nullable State state) {
```

Comment:
Instead of introducing a ton of new utility methods, we should rather improve our code by eliminating nullability for `State`.